src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -0,0 +1,159 @@
+  })
+  const sharedFolder = {
+    value: getString('dashboards.shared'),
+    label: getString('dashboards.sharedFolderTitle')
+  }
+  const [folderListItems, setFolderListItems] = React.useState<SelectOption[]>([sharedFolder])
+
+  const {
+    data: foldersList,
+    error,
+    loading: isFoldersLoading
+  } = useGetFolder({
+    queryParams: { accountId: accountId }
+  })
+
+  function isMissingFolderId(folders: SelectOption[]): boolean {
+    return !folders.some((item: SelectOption) => item.value === formData?.folderId)
+  }
+
+  React.useEffect(() => {
+    const errorResponse = error?.data as ErrorResponse
+    if (errorResponse?.responseMessages) {
+      modalErrorHandler?.showDanger(errorResponse.responseMessages)
+    }
+  }, [error, modalErrorHandler])
+
+  React.useEffect(() => {
+    if (foldersList?.resource) {
+      const folders = foldersList?.resource?.map((folder: { id: string; name: string }) => {
+        return { value: folder?.id, label: folder?.name }
+      })
+      if (isMissingFolderId(folders)) {
+        setInitialValues(prevState => {
+          return { ...prevState, folderId: 'shared' }
+        })
+      }
+      setFolderListItems(prevState => [...prevState, ...folders])
+    }
+  }, [foldersList])
+
+  return (
+    <Layout.Vertical padding="xxlarge">
+      <Heading level={3} font={{ variation: FontVariation.H3 }} padding={{ bottom: 'large' }}>
+        {title}
+      </Heading>
+      <ModalErrorHandler bind={setModalErrorHandler} />
+      {isFoldersLoading ? (
+        <PageSpinner />
+      ) : (
+        <Formik
+          initialValues={initialValues}
+          formName="dashboardForm"
+          validationSchema={Yup.object().shape({
+            folderId: Yup.string().trim().required(getString('dashboards.createFolder.folderNameValidation')),
+            name: Yup.string().trim().required(getString('dashboards.createModal.nameValidation'))
+          })}
+          onSubmit={(completedFormData: DashboardFormikValues) => {
+            onComplete({
+              ...completedFormData,
+              description: completedFormData.description?.join(TAGS_SEPARATOR) || ''
+            })
+          }}
+        >
+          <Form className={css.formContainer}>
+            <Layout.Vertical spacing="large">
+              <FormInput.Select
+                name="folderId"
+                items={folderListItems}
+                label={getString('dashboards.homePage.folder')}
+                placeholder={getString('dashboards.resourceModal.folders')}
+              />
+              <FormInput.Text
+                name="name"
+                label={getString('name')}
+                placeholder={getString('dashboards.createModal.namePlaceholder')}
+              />
+              <FormInput.KVTagInput name="description" label={getString('tagsLabel')} isArray={true} />
+              <Button
+                type="submit"
+                intent="primary"
```

Comment:
You'll also need the button variation.
```suggestion
                intent="primary"
                variation={ButtonVariation.PRIMARY}
```